src/gs-details-page.c

```diff
@@ -57,6 +57,7 @@ struct _GsDetailsPage
 	GsPluginLoader		*plugin_loader;
 	GtkBuilder		*builder;
 	GCancellable		*cancellable;
+	GCancellable		*app_cancellable;
```

Comment:
I'd try to avoid storing this separately, since we store already a reference to the app. It should be enough to call `gs_app_get_cancellable()` where required.

Reply:
You could also add an API like gs_app_get_is_cancelled()... would simplify the code quite a bit not to have to store it tbh.